great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -925,15 +1051,76 @@ def inner_func(
 
                 column_select = [sa.column(column_name) for column_name in column_list]
                 dialect = execution_engine.dialect_module
-                column_function = metric_fn(
+                multicolumn_function = metric_fn(
                     cls,
                     column_select,
                     **metric_value_kwargs,
                     _dialect=dialect,
                     _table=selectable,
                     _metrics=metrics,
                 )
-                return column_function, compute_domain_kwargs, accessor_domain_kwargs
+                return (
+                    multicolumn_function,
+                    compute_domain_kwargs,
+                    accessor_domain_kwargs,
+                )
+
+            return inner_func
+
+        return wrapper
+
+    elif issubclass(engine, SparkDFExecutionEngine):
+        if partial_fn_type is None:
+            partial_fn_type = MetricPartialFunctionTypes.MAP_FN
+        partial_fn_type = MetricPartialFunctionTypes(partial_fn_type)
+        if partial_fn_type != MetricPartialFunctionTypes.MAP_FN:
```

Comment:
style nit: line 1282 uses `not in` for checking the function type; this seems like it should be matched to that.

Reply:
@fjork3 Good point -- we want to keep it as `!=` for now just to indicate that there can only be one.